source/admin_manual/migrating_mailboxes.rst

```diff
@@ -0,0 +1,307 @@
+  #dsync_features = empty-header-workaround
+
+  # Read multiple mails in parallel, improves performance
+  mail_prefetch_count = 20
+
+  # If you have very large mailboxes, you might want to tune this
+  #dsync_commit_msgs_interval = 100 # default in v2.2.30+
+
+  # If you need to hash other headers for whatever reason, you can change this
+  #dsync_hashed_headers = Date Message-ID
+
+  # Avoid useless IO
+  mail_fsync = never
+
+Testing
+=======
+
+It is a very good idea to do some test migrations first using test accounts to ensure everything works.
+
+When migrating mails from another server, you should make sure that these are preserved:
+
+ #. Message flags
+    * Lost flags can be really annoying, you most likely want to avoid it.
+
+ #. Message UIDs and UIDVALIDITY value
+    *  If UIDs are lost, at the minimum clients' message cache gets cleaned and messages are re-downloaded as new.
+    *  Some IMAP clients store metadata by assigning it to specific UID, if UIDs are changed these will be lost.
+
+ #. Mailbox subscription list
+    * Users would be able to manually subscribe them again if you don't want to mess with it.
+
+
+.. _migrating_mailboxes_dovecot:
+
+Migrating mailboxes from another Dovecot
+========================================
+
+Preparations
+------------
+
+If the old system is running v2.1.14+ , you can use doveadm protocol to migrate your mails. If the old system is older, see :ref:`migrating_mailboxes_imapc`.
+
+First, setup doveadm service on old server::
+
+  service doveadm {
+     inet_listener {
+       port = 12354
+     }
+  }
+
+  doveadm_password = supersecret
+
+Then put doveadm password to new server::
+
+  doveadm_password = supersecret
+
+Now you are ready to migrate.
+
+Executing migration
+-------------------
+
+If you want to avoid any changes to your source system, unidirectional synchronization is recommended.
+
+To migrate users, use::
+
+  doveadm sync -1Ru username tcp:host:port
+
+If you are experiencing problems, run::
+
+  doveadm -D sync -1Ru username tcp:host:port
+
+This will enable debug logging.
+
+Note that Public and Shared namespaces are synchronized automatically (see caveats).
+
+It is also possible to do two-way migration if you want to, do it as::
+
+  doveadm backup -Ru username tcp:host:port
+  doveadm sync -u username tcp:host:port
+
+This allows you to do more seamless switchover. Test first!
+
+Caveats
+-------
+
+ * Migrating shared namespaces is inefficient prior v2.3.15+. To migrate Shared namespaces, you need to use -n Shared -n Shared/Mailbox.
+ * If you are using ACLs, some ACL types can cause problems. Especially if you have ACLs that only permit writing to folder, but not reading it. This applies to shared folders only.
+ * Migration can cause unexpectedly high loads on the source system, beware. There is no way to throttle the synchronization.
+ * If you are using LAYOUT=INDEX, ensure you do not access the target mailbox before first run.
```

Comment:
Make this a bit more generic. It's definitely a problem with Maildir INBOX as well. But in LAYOUT=index case it seems to be more of a bug.. How about:

 * Avoid accessing the target mailboxes before the first sync. At best this causes warnings about GUID or UIDVALIDITY changes, and at worst it fails the sync entirely.